repository_service_tuf/helpers/api_client.py

```diff
@@ -88,24 +89,38 @@ def is_logged(settings: LazySettings):
         )
 
 
-def get_headers(settings: LazySettings) -> Dict[str, str]:
-    if settings.get("AUTH") is False:
+def get_headers(
+    settings: LazySettings, token: Optional[str] = None
+) -> Dict[str, str]:
+
+    built_in_auth: bool = settings.get("AUTH")
+
+    if built_in_auth is False and token is None:
         return {}
 
     server = settings.get("SERVER")
-    token = settings.get("TOKEN")
+    token = settings.get("TOKEN") or token
     if server and token:
-        token_access_check = is_logged(settings)
+        additional_error_message: str = ""
+
+        token_access_check = is_logged(settings, token)
         if token_access_check.state is False:
+            if built_in_auth is True:
+                additional_error_message = (
+                    "\n\nTry re-login: 'rstuf --auth admin login'"
+                )
+
```

Comment:
If the status code is 401 or 403, the data is none. I see you handle it in case of built-in auth but not token.
Could you include an else here? 
If  `token_access_check.state` is `False`, and the `built_in_auth` is `False`, the user will see only `None`.
```suggestion
            if built_in_auth is True:
                additional_error_message = (
                    "\n\nTry re-login: 'rstuf --auth admin login'"
                )
            else:
                 additional_error_message = "Token is invalid"

```